File: answer_2/alstamber_ruby/4.rb
def fizzbuzz(n)
  n.times do |i|
    if (i + 1) % 15 == 0
      puts "FizzBuzz"
    elsif (i + 1) % 5 == 0
      puts "Buzz"
    elsif (i + 1) % 3 == 0
      puts "Fizz"
    else
      puts i + 1
    end
  end
end

fizzbuzz(15)

File: answer_2/alstamber_haskell/4.hs
fizzbuzz :: Int -> [String]
fizzbuzz n = map(rp) [1..n]

rp :: Int -> String
rp x | mod x 15 == 0 = "FizzBuzz"
     | mod x 5  == 0 = "Buzz"
     | mod x 3  == 0 = "Fizz"
     | True          = show x

main = mapM_ putStrLn . fizzbuzz . (read::String->Int) =<< getLine
